GliaWidgets/CallVisualizer/VisitorCode/VisitorCodeView.swift

```diff
@@ -0,0 +1,172 @@
+import UIKit
+
+extension CallVisualizer {
+    internal final class VisitorCodeView: BaseView {
+        struct Props: Equatable {
+            enum ViewType: Equatable {
+                case alert(closeButtonTap: Cmd)
+                case embedded
+            }
+
+            let viewType: ViewType
+            let style: AlertStyle
+            let visitorCode: String
+            let isPoweredByShown: Bool
+
+            var isShadowNeeded: Bool {
+                switch viewType {
+                case .alert:
+                    return true
+                case .embedded:
+                    return false
+                }
+            }
+
+            var closeButtonTap: Cmd? {
+                switch viewType {
+                case let .alert(closeButtonTap):
+                    return closeButtonTap
+                case .embedded:
+                    return nil
+                }
+            }
+
+            init(
+                viewType: ViewType = .alert(closeButtonTap: .nop),
+                style: AlertStyle = .initial,
+                visitorCode: String = "11111",
+                isPoweredByShown: Bool = true
+
+            ) {
+                self.viewType = viewType
+                self.style = style
+                self.visitorCode = visitorCode
+                self.isPoweredByShown = isPoweredByShown
+            }
+        }
+
+        var props: Props = Props() {
+            didSet {
+                renderProps()
+            }
+        }
+
+        private let titleLabel = UILabel().make { label in
+            label.numberOfLines = 0
+            label.textAlignment = .center
+        }
+
+        private lazy var stackView = UIStackView.make(.vertical, spacing: 24)(
+            titleLabel,
+            visitorCodeStack,
+            poweredBy
+        )
+
+        private lazy var poweredBy: PoweredBy = {
+            PoweredBy(style: props.style.poweredBy)
+        }()
+
+        private lazy var closeButton = Button(
+            kind: .visitorCodeClose,
+            tap: { [weak self] in
+                self?.props.closeButtonTap?()
+            }
+        ).make { button in
+            button.accessibilityIdentifier = "visitorCode_alert_close_button"
+        }
+
+        private lazy var visitorCodeStack = UIStackView.make(.horizontal, spacing: 8)()
+        private let contentInsets = UIEdgeInsets(top: 24, left: 24, bottom: 21, right: 24)
+        private let cornerRadius: CGFloat = 30
+
+        override func setup() {
+            super.setup()
+            clipsToBounds = true
+            layer.masksToBounds = false
+            layer.cornerRadius = cornerRadius
+            renderProps()
+        }
+
+        override func defineLayout() {
+            super.defineLayout()
+            addSubview(stackView)
+            addSubview(closeButton)
+            visitorCodeStack.distribution = .fill
+            visitorCodeStack.translatesAutoresizingMaskIntoConstraints = false
```

Comment:
I think this line is redundant, since you do it in `make` function ([here](https://github.com/salemove/ios-sdk-widgets/blob/master/GliaWidgets/Layout/UIStackView.Extensions.swift#L47)), does it?
```suggestion
```

Reply:
fixed [676c74c](https://github.com/salemove/ios-sdk-widgets/pull/448/commits/676c74ceb781899d4ca4fa8a8ab684c8a1837308)